### DISCUSSION TITLE
Workflow Transition Guards - If Modified

### DISCUSSION THREAD

**Discussion by cancan101** [pimcore/pimcore#16665] [Q&A / Support]
Is there anyway to write a transition guard that takes into account whether the object was modified while it was in the most current state? For example, if the object was modified by the (current user) in the current state, I want to require it gets sent back for re-review. Otherwise it can be sent onward.

**Comment by dpfaffenbauer** [comment]
Difficult. You could try getting the most recent versions and comparing them.

**Reply by cancan101** [reply]
Is there another way to accomplish this same objective? ie I have an approval process that looks like:
```writer -> editor1 -> editor2 -> published```
where I do want to allow either editor to make changes but if they do, the object should be sent to the start of the workflow. Only in the case that they make no changes can it be sent onward.

**Reply by dpfaffenbauer** [reply]
I would try it like this:

 - Don't allow the users to publish the changes
 - Have a pre-update hook for the DataObject, inject the Request Stack and check what fields have been changed
 - If something has been changed, trigger the Workflow Transition to go back
 - If nothing has changed, keep it at the current state

**Reply by cancan101** [reply]
Taking a step back, are we approaching the problem in the wrong way? Should we be doing something different for how our review process works?

**Reply by cancan101** [reply]
The other idea I had was to introduce additional states:
![image](https://github.com/pimcore/pimcore/assets/51059/2602b319-d054-4d3e-ad8f-ff2601852d59)
In this case, any of the states with padlocks are read-only. If the "editor" wants to edit as opposed to just approving it, they can move the Object to an editable state that can only then transition back to the start of the workflow.

Does this make sense? Is there something simpler that I am missing here?
